src/main/java/uk/gov/justice/digital/nomis/service/transformer/OffenderEventsTransformer.java

```diff
@@ -889,6 +888,17 @@ public String caseNoteEventTypeOf(final uk.gov.justice.digital.nomis.jpa.entity.
         return event.getEventType();
     }
 
+    public Long caseNoteIdOf(final uk.gov.justice.digital.nomis.jpa.entity.OffenderEvent event) {
+        if (event.getEventType().equalsIgnoreCase("CASE_NOTE")) {
+            final var eventData = event.getEventData();
+            final var typePattern = Pattern.compile("(?<=\\bcase_note.{0,4}\\bid.{0,4})(\\w+)");
```

Comment:
this is a bit naff, but is the way the other methods do it.  I am relying on the `id` being the first item in the `case_note` json.  There are other `id`s in the json, hence have to do something naff like this.  See how it goes in t3 :man_shrugging: